compat/os/mingw.h

```diff
@@ -32,5 +32,17 @@
 #include <windows.h>
 #endif
 
+#if HAVE_IO_H
+#include <io.h>
+#endif
+
+#if !HAVE_FSYNC
```

Comment:
Missing documentation of why we provide this wrapper. eg.
```suggestion
// _commit(fd) is API-compatible with fsync(fd) and has equivalent behaviour.
#if !HAVE_FSYNC
```

Reply:
The proposed comment does not add any useful information IMO. It simply states what is already obvious from code itself. I am not going to block this PR if it gains this comment, but I do not recommend adding this comment.

If this fix is (and has to be) MinGW-specific, then I recommend writing about _that_ important implementation caveat (inside the function) instead.

Reply:
I think that the fix applies to both mingw and windows native,
however for now I'm targeting only mingw so I'm fine
leaving it in mingw-specific code.
The _commit (and other similar) APIs are not provided by MinGW, but
by Microsoft (as shown at https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/commit?view=msvc-170)